tests/test_parser.py

```diff
@@ -2180,10 +2180,18 @@ def test_dollar_sub():
     check_xonsh_ast({}, "$(ls)", False)
 
 
-def test_dollar_sub_space():
+def test_dollar_sub_trailing_space():
     check_xonsh_ast({}, "$(ls )", False)
 
 
+def test_dollar_sub_leading_space():
+    check_xonsh_ast({}, "$( ls)", False)
+
+
+def test_dollar_sub_leading_trailing_space():
+    check_xonsh_ast({}, "$( ls )", False)
+
+
```

Comment:
just wondering, are you upto refactor these into parameterizing the check_xonsh_ast calls ?

Reply:
Yeah I think that's a good idea. Would it be better to have a single function for all 6 tests, or two separate functions (3 tests each) - one function for `$()`, the other one for `!()`?

Reply:
I think more will be better. with `pytest.param`, can name the cases too.